packages/edit-navigation/src/components/layout/index.js

```diff
@@ -87,7 +94,9 @@ export default function Layout( { blockEditorSettings } ) {
 						{ ! hasFinishedInitialLoad && <Spinner /> }
 
 						{ hasFinishedInitialLoad && ! hasMenus && (
-							<EmptyState />
+							<EmptyState
+								onCreate={ () => setIsMenuDeleted( false ) }
```

Comment:
Seems unusual that this is needed. What does it do?

Reply:
after deleting a menu an app is in a new state where a `MenuSelector` is being displayed and the `header` is hidden. When a new menu is added a `MenuSelector` is hidden and `header` is displayed.  isMenuDeleted is an indicator informing which state should be presented

Reply:
The entire header shouldn't be hidden. In the mockup the 'Navigation' title is still visible. As is the 'Save' button, but lets ignore that for now.

`<Header>` already has some logic internally for hiding the necessary parts (currently using the `hasMenu` variable).

This is why I think consolidating with `EmptyState` is a better solution. `EmptyState` instead would be renamed to `UnselectedMenuState`.

Instead of using `hasMenu` to hide parts of the header, a variable called `hasUnselectedMenu` is used instead.

The logic becomes nicely consolidated and `EmptyState` doesn't need this extra prop.